ambry-api/src/main/java/com.github.ambry/router/ByteRange.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.router;
+
+/**
+ * Represents a byte range for performing ranged get requests.
+ */
+public class ByteRange {
+  private static final long UNDEFINED_OFFSET = -1;
+
+  private final ByteRangeType type;
+  private final long startOffset;
+  private final long endOffset;
+
+  /**
+   * Construct a range from a start offset to an end offset.
+   * @param startOffset the (inclusive) start byte offset.
+   * @param endOffset the (inclusive) end byte offset.
+   * @return A {@link ByteRange} with the specified offsets.
+   * @throws InvalidByteRangeException
+   */
+  public static ByteRange fromOffsetRange(long startOffset, long endOffset)
+      throws InvalidByteRangeException {
+    if (startOffset < 0 || endOffset < startOffset) {
+      throw new InvalidByteRangeException(startOffset, endOffset, ByteRangeType.OFFSET_RANGE);
```

Comment:
The `ByteRange` will be created by the frontend when request headers are parsed. This isn't in this PR, but it will be bubbled up to a RangeNotSatisfiable error by the frontend.